src/useradd.c

```diff
@@ -2073,6 +2083,37 @@ static void create_home (void)
 			strcat (path, "/");
 			strcat (path, cp);
 			if (access (path, F_OK) != 0) {
+				/* Check if parent directory is BTRFS, fail if requesting
+				   subvolume but no BTRFS. The paths cound be different by the
+				   trailing slash
+				 */
+#if WITH_BTRFS
+				if (subvolflg && (strlen(prefix_user_home) - (int)strlen(path)) <= 1) {
+					char *btrfs_check = strdup(path);
+
+					if (!btrfs_check) {
+						fprintf (stderr,
+						         _("%s: error while duplicating string in BTRFS check %s\n"),
+						         Prog, path);
+						fail_exit (E_HOMEDIR);
+					}
+					btrfs_check[strlen(path) - strlen(cp) - 1] = '\0';
```

Comment:
You're calculating `strlen(path)` twice, no? Wouldn't it make more sense to store it in a variable?

Reply:
Probably done by the compiler, no?

Reply:
yes, but I think it is clearer to do this here instead of adding another variable. We are not going to be saving anything here.

But yes, this can be added if you insist.